libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -284,44 +244,33 @@ void UARTDriver::set_blocking_writes(bool blocking)
  */
 bool UARTDriver::tx_pending()
 {
-    return !BUF_EMPTY(_writebuf);
+    return (_writebuf.available() > 0);
 }
 
 /*
   return the number of bytes available to be read
  */
 uint32_t UARTDriver::available()
 {
-    if (!_initialised) {
-        return 0;
-    }
-    uint16_t _tail;
-    return BUF_AVAILABLE(_readbuf);
+    return _readbuf.available();
 }
 
 /*
   how many bytes are available in the output buffer?
  */
 uint32_t UARTDriver::txspace()
 {
-    if (!_initialised) {
-        return 0;
-    }
-    uint16_t _head;
-    return BUF_SPACE(_writebuf);
+    return _writebuf.space();
 }
 
 int16_t UARTDriver::read()
 {
     uint8_t c;
-    if (!_initialised || _readbuf == NULL) {
-        return -1;
-    }
-    if (BUF_EMPTY(_readbuf)) {
+    if (!_initialised) {
         return -1;
     }
-    c = _readbuf[_readbuf_head];
-    BUF_ADVANCEHEAD(_readbuf, 1);
+    c = _readbuf.peek(0);
+    _readbuf.advance(1);
```

Comment:
What previous explanation? The only one I could find is for when writing, here you read and advance, you are doing exactly the same thing as a `read`.